js/src/utils.js

```diff
@@ -1,28 +1,7 @@
-
-var clone_deep = function(obj) {
-    return JSON.parse(JSON.stringify(obj))
-}
-
-/*
-    Object to avoid infinte loops when two objects van update eachother
-*/
-var Lock = function() {
-    var locked = false;
-    this.with = function(f, context) {
-        locked = true
-        try {
-            f.apply(context, arguments)
-        } finally {
-            locked = false
-        }
-    }
-    this.without = function(f, context) {
-        if(!locked) {
-            f.apply(context, arguments)
-        }
-    }
-}
+var clone_deep = function (obj) {
+    return JSON.parse(JSON.stringify(obj));
+};
```

Comment:
Not sure why I didn't use lodash' cloneDeep here instead

Reply:
Actually, you used `cloneDeep` from lodash everywhere. And `utils.js` is never required. I'll simply remove it.